### PR TITLE
Return the pid during the handshake inside the InitializeResponseArgs

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
closes https://github.com/microsoft/testfx/issues/3211

### CODE REVIEW

File: src/Platform/Microsoft.Testing.Platform/ServerMode/SerializerUtilities.cs [hunk 1]
```diff
@@ -515,10 +516,11 @@ static SerializerUtilities()
 
         Deserializers[typeof(InitializeResponseArgs)] = new ObjectDeserializer<InitializeResponseArgs>(properties =>
         {
+            int processId = GetRequiredPropertyFromJson<int>(properties, JsonRpcStrings.ProcessId);
```
Comment (drognanar) [inline src/Platform/Microsoft.Testing.Platform/ServerMode/SerializerUtilities.cs:519]: is it a required property? or should we handle the case that it's missing or null?

File: src/Platform/Microsoft.Testing.Platform/ServerMode/Json/Json.cs [hunk 2]
```diff
@@ -509,6 +510,7 @@ public Json(Dictionary<Type, JsonSerializer>? serializers = null, Dictionary<Typ
 
         _deserializers[typeof(InitializeResponseArgs)] = new JsonElementDeserializer<InitializeResponseArgs>(
           (json, jsonElement) => new InitializeResponseArgs(
+                  ProcessId: json.Bind<int>(jsonElement, JsonRpcStrings.ProcessId),
```
Comment (drognanar) [inline src/Platform/Microsoft.Testing.Platform/ServerMode/Json/Json.cs:513]: similarly, should we handle the null case?

File: src/Platform/Microsoft.Testing.Platform/ServerMode/SerializerUtilities.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/Platform/Microsoft.Testing.Platform/ServerMode/SerializerUtilities.cs:519]: We force that property in the implementation so we have it always. From the protocol perspective is fine from our implementation I don't have scenario where we cannot get the PID, normal mode(jitter) and AOT let us get it.

File: src/Platform/Microsoft.Testing.Platform/ServerMode/Json/Json.cs (hunk 2 above)
Comment (MarcoRossignoli) [reply src/Platform/Microsoft.Testing.Platform/ServerMode/Json/Json.cs:513]: ditto

**Review by drognanar — APPROVED** [review]

**Pull request merged**